index.js

```diff
@@ -1,65 +1,82 @@
   core.info(`Downloading grype from ${url}`);
-  const zipPath = await cache.downloadTool(url);
+  const zipPath = await tools.downloadTool(url);
   core.debug(`Zip saved to ${zipPath}`);
-  const toolDir = await cache.extractZip(zipPath);
+  const toolDir = await tools.extractZip(zipPath);
   core.debug(`Zip extracted to ${toolDir}`);
-  core.debug(`Grype path is ${path.join(toolDir, grypeBinary)}`);
-  return path.join(toolDir, grypeBinary);
+  const binaryPath = path.join(toolDir, grypeExecutableName);
+  core.debug(`Grype path is ${binaryPath}`);
+  return binaryPath;
 }
 
 function isWindows() {
-  return process.platform == "win32";
+  return process.platform === "win32";
 }
 
+/* download grype and return a path to the executable */
 async function downloadGrype(version) {
-  let url = `https://raw.githubusercontent.com/anchore/grype/main/install.sh`;
-
-  core.debug(`Installing ${version}`);
   if (isWindows()) {
-    // caller expects directory to add to path and join with executable name
-    const exeFilePath = await downloadGrypeWindowsWorkaround(version);
-    core.debug(`Grype saved to ${exeFilePath}`);
-    return path.dirname(exeFilePath);
+    return await downloadGrypeWindowsWorkaround(version);
   }
 
+  const installScriptUrl = `https://raw.githubusercontent.com/anchore/grype/main/install.sh`;
+  core.info(`Downloading grype ${version} via ${installScriptUrl}`);
+
   // TODO: when grype starts supporting unreleased versions, support it here
   // Download the installer, and run
-  const installPath = await cache.downloadTool(url);
-
-  let cmd = `sh ${installPath} -d -b ${installPath}_grype ${version}`;
-  await exec.exec(cmd);
-  let grypePath = `${installPath}_grype/${grypeBinary}`;
-
-  // Cache the downloaded file
-  return cache.cacheFile(grypePath, grypeBinary, grypeBinary, version);
+  const installScriptPath = await tools.downloadTool(installScriptUrl);
+  const installToDir = fs.mkdtempSync(
+    path.join(os.tmpdir(), "grype-download-"),
+  );
+
+  const { stdout, exitCode } = await runCommand("sh", [
+    installScriptPath,
+    "-d",
+    "-b",
+    installToDir,
+    version,
+  ]);
+  if (exitCode !== 0) {
+    core.error("Error installing grype:");
+    core.error(stdout);
+    throw new Error("error installing grype");
+  }
+  return path.join(installToDir, isWindows() ? "grype.exe" : "grype");
 }
 
 async function installGrype(version) {
-  let grypePath = cache.find(grypeBinary, version);
+  core.info(`Installing grype ${version}`);
+
+  let grypePath = tools.find(grypeExecutableName, version);
   if (!grypePath) {
     // Not found, install it
     grypePath = await downloadGrype(version);
+    // Cache the downloaded file, get path to directory
+    grypePath = await tools.cacheFile(
+      grypePath,
+      grypeExecutableName,
+      grypeExecutableName,
+      version,
+    );
   }
-
-  // Add tool to path for this and future actions to use
-  core.addPath(grypePath);
```

Comment:
This is not _technically_ a breaking change, since we did not advertise putting grype on the path, but I suspect it will break a number of users of the action who are expecting this behavior, and if we remove this we should make sure to publish a new major version.